config/main.py

```diff
@@ -544,7 +546,11 @@ def reload(filename, yes, load_sysinfo):
         command = "{} -H -k {} --write-to-db".format(SONIC_CFGGEN_PATH, cfg_hwsku)
         run_command(command, display_cmd=True)
 
-    command = "{} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, filename)
+    if os.path.isfile('/etc/sonic/init_cfg.json'):
+        command = "{} -j {} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, filename, INIT_CFG_FILE)
```

Comment:
Have you confirmed this is the appropriate order of the "-j" arguments to first load the init_cfg.json file followed by the config_db.json file? It looks like sonic-cfggen loads the files in the order they are supplied: https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/sonic-cfggen#L258

Reply:
For the statement of line 550, sonic-cfggen will first load config_db.json and then init_cfg.json, right?

Reply:
That's what it looks like to me. Have you confirmed this? We want to first load init_cf.json, then load config_db.json, overwriting any values which exist in both. Please confirm.

Reply:
I think sonic-cfggen should first load init_cfg.json and then config_db.json.

Reply:
If it loads in the order provided, it would first load config_db.json and then init_cfg.json. Please test and verify the behavior and report back.

Reply:
You just made a commit to change the order, but you haven't reported back your results. Can you please confirm the order by testing?

Reply:
Joe, I tested on DUT using command **_sonic-cfggen -j /etc/sonic/init_cfg.json -j /etc/sonic/config_db.json --write-to-db_**. In init_cfg.json file, the initial values for "auto_restart" and "high_mem_alert" are "disabled". In config_db.json, the initial values for these two keys are "enabled". After running this command and then using command **_show container feature autorestart_**, the key for "auto_restart" will be overrode to "enabled".

Reply:
OK, thanks. Can you also confirm that if you swap the order of the files back to `sonic-cfggen -j /etc/sonic/config_db.json -j /etc/sonic/init_cfg.json --write-to-db` that the results will be wrong, as we assume?

Reply:
Yes, I tested on DUT using the command **_sonic-cfggen -j /etc/sonic/config_db.json -j /etc/sonic/init_cfg.json `--write-to-db_**. Initially the key "auto_restart“ for database in init_cfg.json is set to "disabled" and this key for database is set to "enabled" in config_db.json. After running this command and the command` show container feature autorestart`, the value for auto_restart will be overrode to "disabled".

Reply:
Great. Thanks! I believe we saw at least one place in sonic-buildimage where the order is reversed. When you begin work in there, please check for and fix any instances of this.

Reply:
Yes. for sonic-cfggen command, I found the following locations which we can review and may add init_cfg.json. Some locations already load init_cfg.json.  files/image_config/updategraph/updategraph ===> line 10; files/scripts/configdb-load.sh ===> line 10;  files/build_templates/docker_image_ctl.j2 ===> line 111; files/image_config/config-setup/config-setup ===> line 110 and 190;  I will check and change the order.